mirrord/protocol/src/tcp.rs

```diff
@@ -152,19 +149,36 @@ pub struct InternalHttpResponse {
 pub struct HttpResponse {
     /// This is used to make sure the response is sent in its turn, after responses to all earlier
     /// requests were already sent.
-    pub request_id: u64,
-    pub connection_id: ConnectionId,
     pub port: Port,
+    pub connection_id: ConnectionId,
+    pub request_id: RequestId,
     #[bincode(with_serde)]
-    pub request: InternalHttpResponse,
+    pub response: InternalHttpResponse,
+}
+
+impl PartialOrd<Self> for HttpResponse {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+/// An total order which a request with a LESSER ID is GREATER.
```

Comment:
Couldn't parse this sentence

Reply:
Yeah that's bad, how about:
```suggestion
/// A total order of responses in which the response with the LESSER `request_id` is the GREATER one.
```

Reply:
I don't understand the word 'total' in this context, it's fine otherwise though